tests/inboundAgent.Tests.ps1

```diff
@@ -129,13 +129,8 @@ Describe "[$global:JDK $global:FLAVOR] build args" {
     BeforeAll {
         Push-Location -StackName 'agent' -Path "$PSScriptRoot/.."
         # an old jdk11 image version
-        $TEST_VERSION="4.7"
+        $TEST_VERSION="3063.v26e24490f041"
```

Comment:
This is incorrect, this is supposed to test that the build args work, so they should be different from the default.

Reply:
Why should we test this? It feels weird but I might misunderstand.

(i’ll rollback the changes since the pr was automerged before your review was applied, sorry for that)

Reply:
It's how we test overriding the build args. We use old versions of the agent. Other people use these Dockerfiles so we need to test this works.

Reply:
For Windows, I understand the value of this test. But for Linux images I don't see any value since we already test the build-arg feature when building with docker bake right?

Reply:
The issue is that the test does not freeze the list of tags but assumes the "current" one.

For instance `4.3-6` does not have the new `jdk17` which fails the test.

Also the cost of these tests seems really heavy: having to execute a `docker run` to check if the version is the same as specified could be tested with a `docker inspect` right?

Reply:
I was mainly speaking about the Windows tests, I just noticed the change to Linux and mentioned it as well.  I am not as interested in the Linux tests. The containers are not that heavy to run really, they startup pretty quick.

Reply:
I've tried the following: https://github.com/jenkinsci/docker-inbound-agent/pull/286

Reply:
> The containers are not that heavy to run really, they startup pretty quick.

Then it means that the Windows test harness does not parallelize tests: it takes around 10to 12 min for the "testing" part for each windows image while their build is close to 2 min.

Does pester have a parallel feature, or eventually a powershell loop that would run tests and sync (a bit like `xargs` would do)?

Reply:
Pester does not have a parallel capability. Is this about cost?